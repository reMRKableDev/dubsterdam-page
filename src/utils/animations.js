import { keyframes } from "styled-components";
import { bounce, fadeInUp, fadeInDown, shake, merge } from "react-animations";

export const bounceAnimation = keyframes`${bounce}`;
export const fadeInUpAnimation = keyframes`${fadeInUp}`;
export const fadeInDownAnimation = keyframes`${fadeInDown}`;
export const shakeAnimation = keyframes`${shake}`;
export const fadeInUpShakeAnimation = keyframes`${merge(fadeInUp, shake)}`;
