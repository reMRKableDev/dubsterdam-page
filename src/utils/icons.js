import { Mail, Github, Linkedin, Database } from "@styled-icons/feather";
import {
  Css3,
  Html5,
  ReactLogo,
  Nodejs,
  Javascript,
} from "@styled-icons/boxicons-logos";

export const mailIcon = Mail;
export const gitHubIcon = Github;
export const linkedInIcon = Linkedin;
export const databaseIcon = Database;
export const cssIcon = Css3;
export const htmlIcon = Html5;
export const reactIcon = ReactLogo;
export const nodeIcon = Nodejs;
export const javascriptIcon = Javascript;
