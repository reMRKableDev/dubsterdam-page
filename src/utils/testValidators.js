export const validateNotEmpty = (received) => {
  expect(received).not.toBeNull();
  expect(received).toBeTruthy();
};

export const validateDarkThemeMode = (received) => {
  expect(received).not.toMatch(/light/i);
  expect(received).toMatch(/dark/i);
};

export const validateLightThemeMode = (received) => {
  expect(received).not.toMatch(/dark/i);
  expect(received).toMatch(/light/i);
};
