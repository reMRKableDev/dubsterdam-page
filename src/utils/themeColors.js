import theme from "styled-theming";

export const bgColor = theme("mode", {
  light: "#fafafa",
  dark: "#222",
});

export const footerColor = theme("mode", {
  light: "#222",
  dark: "#fafafa",
});

export const textColor = theme("mode", {
  light: "#000",
  dark: "#fff",
});

export const buttonBgColor = theme("mode", {
  light: "#222",
  dark: "#eee",
});

export const buttonTextColor = theme("mode", {
  light: "#eee",
  dark: "#222",
});

export const brandColor = theme("mode", {
  light: "#ed6c6d",
  dark: "#ed6c6d",
});

export const textAccentColor = theme("mode", {
  light: "#555",
  dark: "#fff",
});
