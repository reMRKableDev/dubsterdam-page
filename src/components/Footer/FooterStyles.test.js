import React from "react";
import { shallow } from "enzyme";

import {
  StyledFooter,
  StyledH2,
  StyledFooterPhrase,
  StyledFooterDeveloperTag,
} from "./FooterStyles";
import { validateMatchSnapshot } from "../../utils/testValidators";

describe("Footer styles unit tests", () => {
  it("should shallow render StyledFooter and match snapshot", () => {
    const wrapper = shallow(<StyledFooter />);
    validateMatchSnapshot(wrapper);
  });

  it("should shallow render StyledH2 and match snapshot", () => {
    const wrapper = shallow(<StyledH2 />);
    validateMatchSnapshot(wrapper);
  });

  it("should shallow render StyledFooterPhrase and match snapshot", () => {
    const wrapper = shallow(<StyledFooterPhrase />);
    validateMatchSnapshot(wrapper);
  });

  it("should shallow render StyledFooterDeveloperTag and match snapshot", () => {
    const wrapper = shallow(<StyledFooterDeveloperTag />);
    validateMatchSnapshot(wrapper);
  });
});
