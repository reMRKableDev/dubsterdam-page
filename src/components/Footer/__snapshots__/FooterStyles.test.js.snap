// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer styles unit tests should shallow render StyledFooter and match snapshot 1`] = `
.c0 {
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 300px;
  z-index: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<footer
  className="c0"
/>
`;

exports[`Footer styles unit tests should shallow render StyledFooterDeveloperTag and match snapshot 1`] = `
.c0 {
  font-family: "Rubik",sans-serif;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

<p
  className="c0"
/>
`;

exports[`Footer styles unit tests should shallow render StyledFooterPhrase and match snapshot 1`] = `
.c0 {
  font-family: "Karla",sans-serif;
  margin-bottom: 25px;
}

<p
  className="c0"
/>
`;

exports[`Footer styles unit tests should shallow render StyledH2 and match snapshot 1`] = `
.c0 {
  font-size: 6vw;
  font-family: "Rubik",sans-serif;
  margin: 1em 0;
  text-transform: uppercase;
}

.c0 span {
  border: 4px solid;
  box-shadow: 0 0 10px;
  padding: 5px;
  text-transform: none;
}

@media (min-width:768px) {
  .c0 {
    font-size: 5.5vw;
    margin: 0.8em 0;
  }

  .c0 span {
    border: 6px solid;
  }
}

@media (min-width:992px) {
  .c0 {
    font-size: 5vw;
    margin: 0.6em 0;
  }

  .c0 span {
    border: 8px solid;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 4.5vw;
    margin: 0.5em 0;
  }

  .c0 span {
    border: 10px solid;
  }
}

<h2
  className="c0"
/>
`;
